apollo-router/src/plugins/telemetry/config_new/graphql/selectors.rs

```diff
@@ -0,0 +1,246 @@
+use schemars::JsonSchema;
+use serde::Deserialize;
+use tower::BoxError;
+
+use crate::plugins::demand_control::cost_calculator::schema_aware_response::TypedValue;
+use crate::plugins::telemetry::config_new::Selector;
+use crate::Context;
+
+#[derive(Deserialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub(crate) enum ListLength {
+    /// The length of the list
+    Value,
+}
+
+#[derive(Deserialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub(crate) enum FieldName {
+    /// The GraphQL field name
+    String,
+}
+
+#[derive(Deserialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub(crate) enum FieldType {
+    /// The GraphQL field name
+    Name,
+    /// The GraphQL field type
+    /// - `bool`
+    /// - `number`
+    /// - `scalar`
+    /// - `object`
+    /// - `list`
+    Type,
+}
+
+#[derive(Deserialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub(crate) enum TypeName {
+    /// The GraphQL type name
+    String,
+}
+
+#[derive(Deserialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields, untagged)]
+pub(crate) enum GraphQLSelector {
+    /// If the field is a list, the length of the list
+    ListLength {
+        #[allow(dead_code)]
+        list_length: ListLength,
+    },
+    /// The GraphQL field name
+    FieldName {
+        #[allow(dead_code)]
+        field_name: FieldName,
+    },
+    /// The GraphQL field type
+    FieldType {
+        #[allow(dead_code)]
+        field_type: FieldType,
+    },
+    /// The GraphQL type name
+    TypeName {
+        #[allow(dead_code)]
+        type_name: TypeName,
+    },
+}
+
+impl Selector for GraphQLSelector {
+    type Request = crate::services::supergraph::Request;
+    type Response = crate::services::supergraph::Response;
+    type EventResponse = crate::graphql::Response;
+
+    fn on_request(&self, _request: &Self::Request) -> Option<opentelemetry::Value> {
+        None
+    }
+
+    fn on_response(&self, _response: &Self::Response) -> Option<opentelemetry::Value> {
+        None
+    }
+
+    fn on_error(&self, _error: &BoxError) -> Option<opentelemetry::Value> {
+        None
+    }
+
+    fn on_response_field(
```

Comment:
I don't know if it's relevant here but please make sure that static attribute is still working with this. In this [PR](https://github.com/apollographql/router/pull/5221/files#diff-f027c8a283d094dd305afc4105b1f483007f683bb298e239a130cffb27cc9964R821-R822) I add to add it at every stages to make sure it happens everytime. For example if you put a condition on an attribute that is executed at the `response_field` stage then you might need to add it. Please add a test like [this one](https://github.com/apollographql/router/pull/5221/files#diff-4e88ae2cd1575d4fe0bf43debbc4e4e8b685f864fdd896e7f6e309915cf85513R589) to make sure it works.